datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -760,6 +766,29 @@ impl<'a, S: SimplifyInfo> ExprRewriter for Simplifier<'a, S> {
                 },
             },
 
+            // unfold the expression round(source, n)
+            Expr::ScalarFunction {
+                fun: BuiltinScalarFunction::Round,
+                args,
```

Comment:
I would prefer that we don't re-write this in the logical plan to work around the fact that we don't currently support this in the physical plan. This makes the logical plan harder to use for other query engines, such as Dask SQL, that are using DataFusion for SQL query planning.

I suggest for this PR that we throw an error in the physical query planner if we see `round` with two arguments. This would be better than producing incorrect results. We can implement the functionality in the physical plan in a separate PR.

Reply:
> This makes the logical plan harder to use for other query engines

Why is that? Could you please give an example?

> We can implement the functionality in the physical plan in a separate PR.

The reason that I don't implement in physical plan is that the implementation isn't straightforward.
Rust only support the `round` function with one argument. 
If we want to implement in physical plan, we have to assemble some arrow compute kernel together, which the logic is same as what I've done in the physical plan. However, the drawback is that, we can't optimize the expression if the second function is a constant, which we have to add additional optimize rules.

The disadvantages of implementing in logical plan are:
1. Put the implementation in the logical optimizer isn't a good solution, we should unfold the expression before doing any optimization.
1. the query plan isn't user-friendly to read because `round` is unfolded. But I think this is not a big deal, as the optimized code is always ugly to read. Users can also use `explain verbose` to find what has happened.

Reply:
> Why is that? Could you please give an example?

Sure. Taking Dask SQL as an example, we would want to translate `round(a, b)` to a call to a Python function to perform the rounding (perhaps in numpy). With this PR, the logical plan now has a complex expression that is specific to the DataFusion implementation. Dask SQL would have to undo the `pow` logic here to get back to the original parameters to `round` so that it can pass them to numpy.

```
Projection: round(CAST(test.column_1 AS Float64) * Float64(0.01)) / Float64(0.01) AS round(test.column_1,Int64(-2))
```